src/values/struct_value.rs

```diff
@@ -23,6 +25,56 @@ impl<'ctx> StructValue<'ctx> {
         }
     }
 
+    /// Gets the value of a field belonging to this `StructType`.
+    ///
+    /// ```no_run
+    /// use inkwell::context::Context;
+    ///
+    /// let context = Context::create();
+    /// let i32_type = context.i32_type();
+    /// let i8_type = context.i8_type();
+    /// let i8_val = i8_type.const_all_ones();
+    /// let i32_val = i32_type.const_all_ones();
+    /// let struct_type = context.struct_type(&[i8_type.into(), i32_type.into()], false);
+    /// let struct_val = struct_type.const_named_struct(&[i8_val.into(), i32_val.into()]);
+    ///
+    /// assert!(struct_val.get_field_at_index(0).is_some());
+    /// assert!(struct_val.get_field_at_index(1).is_some());
+    /// assert!(struct_val.get_field_at_index(3).is_none());
+    /// assert!(struct_val.get_field_at_index(0).unwrap().is_int_value());
+    /// ```
+    pub fn get_field_at_index(self, index: u32) -> Option<BasicValueEnum<'ctx>> {
+        if unsafe { LLVMIsConstant(self.as_value_ref()) } == 0 {
```

Comment:
Why does it need to be constant?

Reply:
Yeah seems to be not needed as LLVMGetNumOperands just returns 0 if they are not accessible.